{"tags": ["DeepLearning", "Python", "brew", "Pylearn2", "\u6a5f\u68b0\u5b66\u7fd2"], "context": " More than 1 year has passed since last update.\n\n\u306f\u3058\u3081\u306b\n\nMacBookAir(2012)\u3067\u6d41\u884c\u308a(?)\u306eDeepLearning\u3068\u304b\u3084\u3063\u3066\u307f\u305f\u304f\u3066\uff0c\u3044\u304f\u3064\u304b\u306e\u8a18\u4e8b\uff08\u53c2\u8003\u6587\u732e\uff09\u3092\u6f01\u3063\u305f\n\u307b\u307c\u53c2\u8003\u6587\u732e\u306e\u30b5\u30a4\u30c8\u306e\u624b\u9806\u3092\u305d\u306e\u307e\u307e\u306a\u305e\u3063\u305f\uff0c\u305d\u306e\u6642\u306b\u3044\u304f\u3064\u304b\u30cf\u30de\u3063\u305f\u5834\u6240\u3082\u3042\u3063\u305f\u306e\u3067\u5099\u5fd8\u9332\u7684\u3068\u3057\u3066\u4fdd\u5b58\n\u30b9\u30af\u30ea\u30d7\u30c8\u3068\u304b\u8a73\u3057\u3044\u5185\u5bb9\u306f\u305d\u3061\u3089\u306e\u30b5\u30a4\u30c8\u3078\n\u3068\u308a\u3042\u3048\u305a\u4f55\u679a\u304b\u753b\u50cf\u3092\u8aad\u307f\u8fbc\u307e\u305b\u3066\u305d\u308c\u3063\u307d\u3044\u753b\u50cf\u304c\u8868\u793a\u3059\u308b\u3068\u3053\u308d\u307e\u3067\u3084\u3063\u3066\u307f\u305f\n\n\n\u74b0\u5883\u69cb\u7bc9\n\nPython,git,pip,numpy,scipy\u3092\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3057\u3066\u304a\u304f\nTheano\u3092\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\n\npip install theano\n\n\nPylearn2\u3092\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\n\npip install pillow\npip install PyYAML\npip install IPython\npip install Cython\n\ngit clone git://github.com/lisa-lab/pylearn2.git\npython setup.py develop\n\n\n\u3053\u3053\u307e\u3067\u3067\u30cf\u30de\u3063\u305f\u3068\u3053\u308d\nImportError: No module named numpy\n\n\nPylearn2\u306esetup.py\u3092\u5b9f\u884c\u3059\u308b\u3068import numpy\u304c\u898b\u3064\u304b\u3089\u306a\u3044\u7cfb?\n\n\n\u4ee5\u4e0b\u3092\u53c2\u8003\u306b\u5bfe\u51e6\u3057\u3066\u307f\u305f\u304c\u3046\u307e\u304f\u6cbb\u3089\u305a \nhttp://d.hatena.ne.jp/aremokoremo/20140507/1399475248\n\u74b0\u5883\u5909\u6570(~/.bash_profile)\u3068\u304b\u78ba\u8a8d\u3057\u3066path\u3092\u78ba\u8a8d\u3057\u305f\u308a\u3057\u305f\n\u5fdc\u6025\u51e6\u7f6e\u7684\u306b\u3068\u308a\u3042\u3048\u305anumpy\u304c\u7f6e\u3044\u3066\u3042\u308bsite-packages\u306e\u30d1\u30b9\u3092\u8ffd\u52a0\n\u74b0\u5883\u5909\u6570\u3092\u518d\u8aad\u307f\u8fbc\u307f(source ~/.bash_profile)\n\n\n\n===============================\n00001   #include <Python.h>\n00002   #include \"structmember.h\"\n00003   #include <sys/time.h>\n~~~~~~~~~\u7565~~~~~~~~~\nException: Compilation failed (return status=1): /Users/user-name/.theano/compiledir_Darwin-14.5.0-x86_64-i386-64bit-i386-2.7.8-64/lazylinker_ext/mod.cpp:1:10: fatal error: 'Python.h' file not found. #include <Python.h>.          ^. 1 error generated.. \n\n\nimport theano\u3067\u30a8\u30e9\u30fc Python.h\u3068\u304b\u304c\u898b\u3064\u304b\u3089\u306a\u3044\u7cfb\uff1f\n\n\n\u3044\u3088\u3044\u3088\u74b0\u5883\u304c\u58ca\u308c\u3066\u3044\u308b\u3088\u3046\u306a\u6c17\u304c\u3057\u3066\u304d\u305f\u306e\u3067brew doctor\u3067\u78ba\u8a8d\n\u30a8\u30e9\u30fc\u5185\u5bb9\u3092\u4e00\u3064\u4e00\u3064\u5730\u9053\u306b\u6d88\u3059\uff08\u30a8\u30e9\u30fc\u5185\u5bb9\u3092Google\u5148\u751f\u306b\u805e\u304f\u3068\u89e3\u6c7a\u7b56\u304c\u3042\u308b\n\u3064\u3044\u3067\u306bbrew update\u3068\u304b\u3057\u305f\n\n\n\n\n\u5b9f\u884c\n\n\u624b\u9806\n\n\n\u540c\u3058\u30b5\u30a4\u30ba\u306e\u753b\u50cf\u3092\u96c6\u3081\u3066\u304f\u308b\n\nGoogle\u304b\u3089\u5b66\u7fd2\u3055\u305b\u305f\u3044\uff0cjpg\u306e\u753b\u50cf\u3092\u62fe\u3063\u3066\u304f\u308b(10\u679a\u304f\u3089\u3044)\n\n\n\n\u30b9\u30af\u30ea\u30d7\u30c8\u3092\u4f7f\u3063\u3066\u30b0\u30ec\u30fc\u30b9\u30b1\u30fc\u30eb\u5316 & csv\u306b\u3059\u308b\n\n\u30b5\u30a4\u30c8\u306b\u3042\u308bimg2csv.py\u3092\u5b9f\u884c\u3059\u308b\nImage\u30e2\u30b8\u30e5\u30fc\u30eb\u3068\u304b\u304c\u898b\u3064\u304b\u3089\u306a\u3044\u3068\u304b\u8a00\u308f\u308c\u305f\u30892\u884c\u76ee\u304b\u30894\u884c\u76ee\u3092\u4ee5\u4e0b\u306b\u5909\u66f4\n\nfrom PIL import Image\nfrom PIL import ImageOps\nfrom PIL import ImageFilter\n\n\n\u30b5\u30a4\u30c8\u306e\u901a\u308a\u4ee5\u4e0b\u306e\u30b3\u30de\u30f3\u30c9\u3092\u5b9f\u884c\n\npython img2csv.py \u30b0\u30ec\u30fc\u30b9\u30b1\u30fc\u30eb\u5909\u63db\u3057\u305f.jpg\u306e\u30ea\u30b9\u30c8 train.csv\n\n\n\u3053\u3053\u3067\u3046\u307e\u304f\u30ea\u30b9\u30c8\u304c\u6e21\u305b\u306a\u304b\u3063\u305f\u306e\u3067\u30b3\u30fc\u30c9\u3092\u8aad\u3093\u3067\u4ee5\u4e0b\u3067\u5b9f\u884c\n\npython img2csv.py \"../img/*.jpg\" train.csv\n\n\n\n\u5b66\u7fd2\u3055\u305b\u308b\n\npylearn2/datasets/\u306b\u30b5\u30a4\u30c8\u306ehoge_dataset.py\u3092\u5165\u308c\u308b\n\u74b0\u5883\u5909\u6570\u306b\u4ee5\u4e0b\u3092\u8a2d\u5b9a\n\nexport PYLEARN2_DATA_PATH=train.csv\u3068grbm.yaml\u304c\u7f6e\u3044\u3066\u3042\u308b\u30d1\u30b9\nexport PYLEARN2_VIEWER_COMMAND=\"open -Wn\"\n\n\n\n\u5b66\u7fd2\u3055\u305b\u308b\ntrain.py grbm.yaml\n\n\n\n\n\n\u5b66\u7fd2\u306e\u7d50\u679c\u3092\u8868\u793a\u3059\u308b\n\n\n\u8868\u793a\u306e\u30b3\u30de\u30f3\u30c9\nshow_weights.py grbm.pkl\n\n\n\n\n\n\n\u3053\u3053\u307e\u3067\u3067\u30cf\u30de\u3063\u305f\u3068\u3053\u308d\n\n\u5b66\u7fd2\u6642\u9593\n\n\u601d\u3063\u305f\u3088\u308a\u3082\u6642\u9593\u304c\u304b\u304b\u308b\n150*150\u306e\u753b\u50cf\u309210\u679a\u4f7f\u3063\u3066\u5b9f\u884c\u3057\u306620\u5206\u3050\u3089\u3044\u304b\u304b\u3063\u305f\n\n\n\u5b66\u7fd2\u7d50\u679c\u306e\u8868\u793a\uff08\u63cf\u753b\uff09\n\n\n\u5b66\u7fd2\u7d50\u679c\u3092\u8868\u793a\u3059\u308b\u6642\u306b\u4ee5\u4e0b\u306e\u3088\u3046\u306a\u30a8\u30e9\u30fc\u304c\u51fa\u305f\nUserWarning: Unable to import matplotlib. Some features unavailable. Original exception: No module named matplotlib.pyplot\n\"Original exception: \" + str(matplotlib_exception))\n\n\n\n\u4ee5\u4e0b\u306e\u30b3\u30de\u30f3\u30c9\u3067matplotlib\u3092\u8ffd\u52a0\u3057\u3066\u7121\u4e8b\u306b\u52d5\u4f5c\u3092\u78ba\u8a8d\npip install matplotlib\n\n\n\n\n\u53c2\u8003\u6587\u732e\nhttp://kensuke-mi.xyz/kensuke-mi_diary/2014/11/rbma.html\nhttp://d.hatena.ne.jp/aremokoremo/20140507/1399475248\n# \u306f\u3058\u3081\u306b\n* MacBookAir(2012)\u3067\u6d41\u884c\u308a(?)\u306eDeepLearning\u3068\u304b\u3084\u3063\u3066\u307f\u305f\u304f\u3066\uff0c\u3044\u304f\u3064\u304b\u306e\u8a18\u4e8b\uff08\u53c2\u8003\u6587\u732e\uff09\u3092\u6f01\u3063\u305f\n* \u307b\u307c\u53c2\u8003\u6587\u732e\u306e\u30b5\u30a4\u30c8\u306e\u624b\u9806\u3092\u305d\u306e\u307e\u307e\u306a\u305e\u3063\u305f\uff0c\u305d\u306e\u6642\u306b\u3044\u304f\u3064\u304b\u30cf\u30de\u3063\u305f\u5834\u6240\u3082\u3042\u3063\u305f\u306e\u3067\u5099\u5fd8\u9332\u7684\u3068\u3057\u3066\u4fdd\u5b58\n* \u30b9\u30af\u30ea\u30d7\u30c8\u3068\u304b\u8a73\u3057\u3044\u5185\u5bb9\u306f\u305d\u3061\u3089\u306e[\u30b5\u30a4\u30c8](http://kensuke-mi.xyz/kensuke-mi_diary/2014/11/rbma.html)\u3078\n* \u3068\u308a\u3042\u3048\u305a\u4f55\u679a\u304b\u753b\u50cf\u3092\u8aad\u307f\u8fbc\u307e\u305b\u3066\u305d\u308c\u3063\u307d\u3044\u753b\u50cf\u304c\u8868\u793a\u3059\u308b\u3068\u3053\u308d\u307e\u3067\u3084\u3063\u3066\u307f\u305f\n\n# \u74b0\u5883\u69cb\u7bc9\n* Python,git,pip,numpy,scipy\u3092\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\u3057\u3066\u304a\u304f\n* Theano\u3092\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\n\n```\u30b3\u30de\u30f3\u30c9:\npip install theano\n```\n* Pylearn2\u3092\u30a4\u30f3\u30b9\u30c8\u30fc\u30eb\n\n```\u30b3\u30de\u30f3\u30c9:\npip install pillow\npip install PyYAML\npip install IPython\npip install Cython\n\ngit clone git://github.com/lisa-lab/pylearn2.git\npython setup.py develop\n```\n\n### \u3053\u3053\u307e\u3067\u3067\u30cf\u30de\u3063\u305f\u3068\u3053\u308d\n```\u30b3\u30de\u30f3\u30c9:\nImportError: No module named numpy\n```\n\n* Pylearn2\u306esetup.py\u3092\u5b9f\u884c\u3059\u308b\u3068import numpy\u304c\u898b\u3064\u304b\u3089\u306a\u3044\u7cfb?\n    * \u4ee5\u4e0b\u3092\u53c2\u8003\u306b\u5bfe\u51e6\u3057\u3066\u307f\u305f\u304c\u3046\u307e\u304f\u6cbb\u3089\u305a \n    * http://d.hatena.ne.jp/aremokoremo/20140507/1399475248\n    * \u74b0\u5883\u5909\u6570(~/.bash_profile)\u3068\u304b\u78ba\u8a8d\u3057\u3066path\u3092\u78ba\u8a8d\u3057\u305f\u308a\u3057\u305f\n    * \u5fdc\u6025\u51e6\u7f6e\u7684\u306b\u3068\u308a\u3042\u3048\u305anumpy\u304c\u7f6e\u3044\u3066\u3042\u308bsite-packages\u306e\u30d1\u30b9\u3092\u8ffd\u52a0\n    * \u74b0\u5883\u5909\u6570\u3092\u518d\u8aad\u307f\u8fbc\u307f(source ~/.bash_profile)\n\n```\u30b3\u30de\u30f3\u30c9:\n===============================\n00001\t#include <Python.h>\n00002\t#include \"structmember.h\"\n00003\t#include <sys/time.h>\n~~~~~~~~~\u7565~~~~~~~~~\nException: Compilation failed (return status=1): /Users/user-name/.theano/compiledir_Darwin-14.5.0-x86_64-i386-64bit-i386-2.7.8-64/lazylinker_ext/mod.cpp:1:10: fatal error: 'Python.h' file not found. #include <Python.h>.          ^. 1 error generated.. \n```\n* import theano\u3067\u30a8\u30e9\u30fc Python.h\u3068\u304b\u304c\u898b\u3064\u304b\u3089\u306a\u3044\u7cfb\uff1f\n    * \u3044\u3088\u3044\u3088\u74b0\u5883\u304c\u58ca\u308c\u3066\u3044\u308b\u3088\u3046\u306a\u6c17\u304c\u3057\u3066\u304d\u305f\u306e\u3067brew doctor\u3067\u78ba\u8a8d\n    * \u30a8\u30e9\u30fc\u5185\u5bb9\u3092\u4e00\u3064\u4e00\u3064\u5730\u9053\u306b\u6d88\u3059\uff08\u30a8\u30e9\u30fc\u5185\u5bb9\u3092Google\u5148\u751f\u306b\u805e\u304f\u3068\u89e3\u6c7a\u7b56\u304c\u3042\u308b\n    * \u3064\u3044\u3067\u306bbrew update\u3068\u304b\u3057\u305f\n\n# \u5b9f\u884c\n## \u624b\u9806  \n\n1. \u540c\u3058\u30b5\u30a4\u30ba\u306e\u753b\u50cf\u3092\u96c6\u3081\u3066\u304f\u308b\n    * Google\u304b\u3089\u5b66\u7fd2\u3055\u305b\u305f\u3044\uff0cjpg\u306e\u753b\u50cf\u3092\u62fe\u3063\u3066\u304f\u308b(10\u679a\u304f\u3089\u3044)\n        \n2. \u30b9\u30af\u30ea\u30d7\u30c8\u3092\u4f7f\u3063\u3066\u30b0\u30ec\u30fc\u30b9\u30b1\u30fc\u30eb\u5316 & csv\u306b\u3059\u308b\n    * \u30b5\u30a4\u30c8\u306b\u3042\u308bimg2csv.py\u3092\u5b9f\u884c\u3059\u308b\n    * Image\u30e2\u30b8\u30e5\u30fc\u30eb\u3068\u304b\u304c\u898b\u3064\u304b\u3089\u306a\u3044\u3068\u304b\u8a00\u308f\u308c\u305f\u30892\u884c\u76ee\u304b\u30894\u884c\u76ee\u3092\u4ee5\u4e0b\u306b\u5909\u66f4\n         \n    ```\u30b3\u30de\u30f3\u30c9:\n    from PIL import Image\n    from PIL import ImageOps\n    from PIL import ImageFilter\n    ```\n    * \u30b5\u30a4\u30c8\u306e\u901a\u308a\u4ee5\u4e0b\u306e\u30b3\u30de\u30f3\u30c9\u3092\u5b9f\u884c\n\n    ```\u30b3\u30de\u30f3\u30c9:\n    python img2csv.py \u30b0\u30ec\u30fc\u30b9\u30b1\u30fc\u30eb\u5909\u63db\u3057\u305f.jpg\u306e\u30ea\u30b9\u30c8 train.csv\n    ```\n    * \u3053\u3053\u3067\u3046\u307e\u304f\u30ea\u30b9\u30c8\u304c\u6e21\u305b\u306a\u304b\u3063\u305f\u306e\u3067\u30b3\u30fc\u30c9\u3092\u8aad\u3093\u3067\u4ee5\u4e0b\u3067\u5b9f\u884c\n              \n    ```\u30b3\u30de\u30f3\u30c9:\n    python img2csv.py \"../img/*.jpg\" train.csv\n    ```\n3. \u5b66\u7fd2\u3055\u305b\u308b\n    * pylearn2/datasets/\u306b\u30b5\u30a4\u30c8\u306ehoge_dataset.py\u3092\u5165\u308c\u308b\n    * \u74b0\u5883\u5909\u6570\u306b\u4ee5\u4e0b\u3092\u8a2d\u5b9a  \n        * export PYLEARN2_DATA_PATH=train.csv\u3068grbm.yaml\u304c\u7f6e\u3044\u3066\u3042\u308b\u30d1\u30b9\n        * export PYLEARN2_VIEWER_COMMAND=\"open -Wn\"\n    * \u5b66\u7fd2\u3055\u305b\u308b\n        \n        ```\u30b3\u30de\u30f3\u30c9: \n        train.py grbm.yaml\n        ```\n4. \u5b66\u7fd2\u306e\u7d50\u679c\u3092\u8868\u793a\u3059\u308b\n    * \u8868\u793a\u306e\u30b3\u30de\u30f3\u30c9\n    \n        ```\u30b3\u30de\u30f3\u30c9: \n        show_weights.py grbm.pkl\n        ```    \n\n## \u3053\u3053\u307e\u3067\u3067\u30cf\u30de\u3063\u305f\u3068\u3053\u308d\n\n### \u5b66\u7fd2\u6642\u9593  \n* \u601d\u3063\u305f\u3088\u308a\u3082\u6642\u9593\u304c\u304b\u304b\u308b\n* 150*150\u306e\u753b\u50cf\u309210\u679a\u4f7f\u3063\u3066\u5b9f\u884c\u3057\u306620\u5206\u3050\u3089\u3044\u304b\u304b\u3063\u305f\n\n### \u5b66\u7fd2\u7d50\u679c\u306e\u8868\u793a\uff08\u63cf\u753b\uff09\n* \u5b66\u7fd2\u7d50\u679c\u3092\u8868\u793a\u3059\u308b\u6642\u306b\u4ee5\u4e0b\u306e\u3088\u3046\u306a\u30a8\u30e9\u30fc\u304c\u51fa\u305f\n\n    ```\u30b3\u30de\u30f3\u30c9:\nUserWarning: Unable to import matplotlib. Some features unavailable. Original exception: No module named matplotlib.pyplot\n  \"Original exception: \" + str(matplotlib_exception))\n    ```\n\n* \u4ee5\u4e0b\u306e\u30b3\u30de\u30f3\u30c9\u3067matplotlib\u3092\u8ffd\u52a0\u3057\u3066\u7121\u4e8b\u306b\u52d5\u4f5c\u3092\u78ba\u8a8d\n    \n    ```\u30b3\u30de\u30f3\u30c9:\n    pip install matplotlib\n    ```\n\n# \u53c2\u8003\u6587\u732e\n[http://kensuke-mi.xyz/kensuke-mi_diary/2014/11/rbma.html](http://kensuke-mi.xyz/kensuke-mi_diary/2014/11/rbma.html)\n[http://d.hatena.ne.jp/aremokoremo/20140507/1399475248](http://d.hatena.ne.jp/aremokoremo/20140507/1399475248)\n"}